ui/src/registerExperimentationService.ts

```diff
@@ -0,0 +1,95 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as tas from 'vscode-tas-client';
+import { IActionContext, IExperimentationServiceAdapter, registerTelemetryHandler } from '../index';
+import { IInternalTelemetryReporter } from './createTelemetryReporter';
+import { ext } from './extensionVariables';
+import { getPackageInfo } from './getPackageInfo';
+
+export async function registerExperimentationService(ctx: vscode.ExtensionContext, targetPopulation?: tas.TargetPopulation): Promise<void> {
+    const result: ExperimentationServiceAdapter = new ExperimentationServiceAdapter();
+    const { extensionId, extensionVersion } = getPackageInfo(ctx);
+
+    if (vscode.workspace.getConfiguration('telemetry').get('enableTelemetry', false)) {
+        try {
+            result.wrappedExperimentationService = await tas.getExperimentationServiceAsync(
+                extensionId,
+                extensionVersion,
+                targetPopulation ?? (/alpha/ig.test(extensionVersion) ? tas.TargetPopulation.Insiders : tas.TargetPopulation.Public),
+                new ExperimentationTelemetry(ext._internalReporter, ctx),
+                ctx.globalState
+            );
+        } catch {
+            // Best effort
+        }
+    }
+
+    ext.experimentationService = result;
+}
+
+class ExperimentationServiceAdapter implements IExperimentationServiceAdapter {

```

Comment:
Consider moving this and `ExperimentationTelemetry` to another file and/or give them their own folder. @EricJizbaMSFT what do you think?

Reply:
No preference

Reply:
I'll leave them here then, keeps everything together with fewer exports.